Cubical/Categories/Displayed/AdjointToReindex.agda

```diff
@@ -0,0 +1,173 @@
+{-# OPTIONS --safe #-}
+--
+module Cubical.Categories.Displayed.AdjointToReindex where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Path
+open import Cubical.Foundations.HLevels
+open import Cubical.Foundations.Isomorphism
+open import Cubical.Data.Sigma
+
+open import Cubical.Categories.Category.Base
+open import Cubical.Categories.Category.Properties
+open import Cubical.Categories.Instances.Terminal
+open import Cubical.Categories.Constructions.BinProduct
+  renaming (Fst to FstBP ; Snd to SndBP)
+open import Cubical.Categories.Functor
+
+open import Cubical.Categories.Displayed.Base
+open import Cubical.Categories.Displayed.Properties
+open import Cubical.Categories.Displayed.Functor
+open import Cubical.Categories.Displayed.Instances.Terminal
+
+private
+  variable
+    ℓC ℓC' ℓCᴰ ℓCᴰ' ℓD ℓD' ℓDᴰ ℓDᴰ' ℓE ℓE' ℓEᴰ ℓEᴰ' : Level
+
+open Categoryᴰ
+
+module _
+  {C : Category ℓC ℓC'} {D : Category ℓD ℓD'}
+  (Cᴰ : Categoryᴰ C ℓDᴰ ℓDᴰ') (F : Functor C D)
+  where
+
+  open Category
+  open Functor F
+  private
+    module Cᴰ = Categoryᴰ Cᴰ
+    open Iso
+
+    hom-path : ∀ {A B A' B'} (p : A ≡ A') (q : B ≡ B') →
+      (D [ A , B ]) ≡ (D [ A' , B' ])
+    hom-path p q = cong₂ (λ a b → D [ a , b ]) p q
+
+    hom-pathP : ∀ {A B A' B'} (p : A ≡ A') (q : B ≡ B') →
+                (f : D [ A , B ]) → (f' : D [ A' , B' ]) →
+                Type ℓD'
+    hom-pathP p q f f' = PathP (λ i → hom-path p q i) f f'
+
+    isPropHomP : ∀ {A B A' B'} (p : A ≡ A') (q : B ≡ B') →
+                (f : D [ A , B ]) → (f' : D [ A' , B' ]) →
+                isProp (hom-pathP p q f f')
+    isPropHomP pdom pcod f f'' x y =
+      isoFunInjective (PathPIsoPath _ _ _) x y fromPathPeq
+      where
+      fromPathPeq : fromPathP x ≡ fromPathP y
+      fromPathPeq = D .isSetHom _ _ (fromPathP x) (fromPathP y)
+
+  ∃F : Categoryᴰ D (ℓ-max (ℓ-max ℓC ℓD) ℓDᴰ) (ℓ-max (ℓ-max ℓC' ℓD') ℓDᴰ')
+  ∃F .ob[_] d = Σ[ c ∈ C .ob ] (F-ob c ≡ d) × Cᴰ.ob[ c ]
+  ∃F .Hom[_][_,_] f (c , p , x) (c' , p' , x') =
+    Σ[ g ∈ C [ c , c' ] ] hom-pathP p p' (F-hom g) f × Cᴰ.Hom[ g ][ x , x' ]
+  ∃F .idᴰ {d} {c , p , x} =
+    (C .id) ,
+    (F-id ◁ (cong (λ v → D .id {v}) p)) ,
+    Cᴰ .idᴰ
+  ∃F ._⋆ᴰ_ (g , p , gᴰ) (h , q , hᴰ) =
+      g ⋆⟨ C ⟩ h ,
+      F-seq g h ◁ (λ i → p i ⋆⟨ D ⟩ q i) ,
+      (Cᴰ._⋆ᴰ_ gᴰ hᴰ)
+  ⋆IdLᴰ ∃F {d}{d'}{g}{c , pc , xc}{c' , pc' , xc'} (f , p , x) =
```

Comment:
These proofs can probably be simplified using https://github.com/agda/cubical/blob/master/Cubical/Categories/Displayed/Reasoning.agda